### DISCUSSION TITLE
crazyflie keeps indicating that it is charging even after being unplugged

### DISCUSSION THREAD

**Discussion by MilesZew** [bitcraze/discussions#518] [Q&A]
I noticed today that my crazyflie was taking an abnormally long time to charge (roughly 2 hours), so I unplugged it, and M3 kept blinking indicating that it was charging, even though there was no way it could have been. I then checked the battery voltage and it said it was around 4.1 volts, which looked about right for fully charged. This makes me think that the indicator just didn't stop blinking and that the crazyflie had been fully charged for a while. 

This problem only started happening after doing the firmware update, which I'm pretty sure was the latest version (I ran the update all projects script on the VM). Does anyone have an idea of what's going on here?

**Comment by hmllr** [comment]
This seems more like a battery aging issue - do you have a newer battery with which you can try it?
For me the LED works as normal with the newest firmware. 
Be aware that the fully charged state is not determined directly by a specific voltage, but by if the charging IC indicates fully charged or not (if you are interested in the code: https://github.com/bitcraze/crazyflie2-nrf-firmware/blob/c3fc80fe1f2a94e65bed0ca4836d0b8270ae0fe1/src/pm.c) - so the LED doesn't necessarily need to fully light up at a deterministic voltage.
Hope this helps,
Hanna

**Reply by MilesZew** [reply]
Thanks Hanna - I only just recently got my crazyflie (around a few weeks ago), so I definitely wouldn't expect there to be battery aging issues yet, unless me doing a some flights (around 20) before doing the firmware update somehow caused a problem

However, regardless I think the led should stop blinking when the crazyflie is unplugged even if there are battery issues, right?

that's good to know about the fully charged state, although I do believe voltage can be used as a decent ballpark value for charge level

**Reply by hmllr** [reply]
indeed, yes, if you unplug it it should anyway stop blinking. 
Did you also update the nRF firmware? Because that's the uC responsible for power management and the only idea I have right now if it was a software issue would be that you have incompatible versions. So maybe try to update it as well (https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/userguides/userguide_client/) and see if this helps.

**Reply by MilesZew** [reply]
Hmm, when I originally did the firmware update I did it as described in the link you posted - however just to make sure there wasn't some sort of fluke I did the firmware update again, and had the same problem. I then tried changing the firmware to a much older version, and no longer had the problem, so I decided to try and pinpoint at exactly what version the error starts happening and I found that with the 2022.05 version of the firmware there was no problem, but with the 2022.09 version, the error was there again. 

As I mentioned I only just recently got my crazyflie so I wouldn't expect there to be any sort of hardware incompatibility, which makes me think that there's some weird bug. 

I took a look at the release notes and noticed this (https://github.com/bitcraze/crazyflie-firmware/issues/1043) issue which seemed suspect - and followed it to this (https://github.com/bitcraze/crazyflie2-nrf-firmware/pull/70) PR, however I don't really know how to investigate further since I'm not at all familiar with the code base, or C.

I think I should open a bug report, unless you have any other ideas about what could be happening

**Reply by knmcguire** [reply]
And what does your battery level in the cfclient say?

Anyway, I'm unfortunately not able to recreate the issue with flashing the latest release 2022.12 with the [cfclient](https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/userguides/userguide_client/#firmware-upgrade).

Before anything I think it would be good to try to flash the latest master  of both the STM and NRF first, so completely from source

- For the STM: https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/building-and-flashing/build/
- For the NRF: https://www.bitcraze.io/documentation/repository/crazyflie2-nrf-firmware/master/build/build/

Since you are using the VM you probably don't have to install the toolchain.

But if you are afraid of breaking anything, and your crazyflie is working, then perhaps it would be good to just ignore it for now. We can keep our eye out and if anybody sees this with the latest master firmwares too we can make a bug report.
